ClosedXML/Excel/XLWorkbook.cs

```diff
@@ -1,884 +1,959 @@
+            }
+            return columns;
+        }
+
+#region Fields
+
+        private readonly XLLoadSource _loadSource = XLLoadSource.New;
+        private readonly String _originalFile;
+        private readonly Stream _originalStream;
+
+#endregion
+
+#region Constructor
+
+
+        /// <summary>
+        ///   Creates a new Excel workbook.
+        /// </summary>
+        public XLWorkbook()
+            :this(XLEventTracking.Enabled)
+        {
+
+        }
+
+        public XLWorkbook(XLEventTracking eventTracking)
+        {
+            EventTracking = eventTracking;
+            DefaultRowHeight = 15;
+            DefaultColumnWidth = 8.43;
+            Style = new XLStyle(null, DefaultStyle);
+            RowHeight = DefaultRowHeight;
+            ColumnWidth = DefaultColumnWidth;
+            PageOptions = DefaultPageOptions;
+            Outline = DefaultOutline;
+            Properties = new XLWorkbookProperties();
+            CalculateMode = XLCalculateMode.Default;
+            ReferenceStyle = XLReferenceStyle.Default;
+            InitializeTheme();
+            ShowFormulas = DefaultShowFormulas;
+            ShowGridLines = DefaultShowGridLines;
+            ShowOutlineSymbols = DefaultShowOutlineSymbols;
+            ShowRowColHeaders = DefaultShowRowColHeaders;
+            ShowRuler = DefaultShowRuler;
+            ShowWhiteSpace = DefaultShowWhiteSpace;
+            ShowZeros = DefaultShowZeros;
+            RightToLeft = DefaultRightToLeft;
+            WorksheetsInternal = new XLWorksheets(this);
+            NamedRanges = new XLNamedRanges(this);
+            CustomProperties = new XLCustomProperties(this);
+            ShapeIdManager = new XLIdManager();
+            Author = Environment.UserName;
+        }
+
+        /// <summary>
+        ///   Opens an existing workbook from a file.
+        /// </summary>
+        /// <param name = "file">The file to open.</param>
+        public XLWorkbook(String file)
+            : this(file, XLEventTracking.Enabled)
+        {
+
+        }
+
+        public XLWorkbook(String file, XLEventTracking eventTracking)
+            : this(eventTracking)
+        {
+            _loadSource = XLLoadSource.File;
+            _originalFile = file;
+            _spreadsheetDocumentType = GetSpreadsheetDocumentType(_originalFile);
+            Load(file);
+        }
+
+
+
+        /// <summary>
+        ///   Opens an existing workbook from a stream.
+        /// </summary>
+        /// <param name = "stream">The stream to open.</param>
+        public XLWorkbook(Stream stream):this(stream, XLEventTracking.Enabled)
+        {
+
+        }
+
+        public XLWorkbook(Stream stream, XLEventTracking eventTracking)
+            : this(eventTracking)
+        {
+            _loadSource = XLLoadSource.Stream;
+            _originalStream = stream;
+            Load(stream);
+        }
+
+#endregion
+
+#region Nested type: UnsupportedSheet
+
+        internal
```

Comment:
Where did you get the hashing algorithm? Maybe add a comment with the link. I also think this should rather be in `XLHelper`.